lib/mixpanel-ruby/consumer.rb

```diff
@@ -88,7 +94,11 @@ def send(type, message)
       form_data = {"data" => data, "verbose" => 1}
       form_data.merge!("api_key" => api_key) if api_key
 
-      response_code, response_body = request(endpoint, form_data)
+      begin
+        response_code, response_body = request(endpoint, form_data)
+      rescue Exception => e
+        raise ConnectionError.new("Could not connect to Mixpanel, with error \"#{e.message}\".")
```

Comment:
[Rescue StandardError, Not Exception](https://robots.thoughtbot.com/rescue-standarderror-not-exception)